$NetBSD: patch-xpcom_reflect_xptcall_src_md_unix_moz.build,v 1.6 2014/05/28 07:18:28 pho Exp $

* Support NetBSD/amd64 and DragonFly/amd64.

* OS_TEST can never be powerpc.

--- xpcom/reflect/xptcall/md/unix/moz.build.orig	2014-05-06 22:56:39.000000000 +0000
+++ xpcom/reflect/xptcall/md/unix/moz.build
@@ -9,7 +9,7 @@ if CONFIG['OS_ARCH'] == 'Darwin':
         'xptcinvoke_darwin.cpp',
         'xptcstubs_darwin.cpp',
     ]
-    if CONFIG['OS_TEST'] == 'powerpc':
+    if CONFIG['OS_TEST'] == 'ppc':
         SOURCES += [
             'xptcinvoke_asm_ppc_rhapsody.s',
         ]
@@ -19,14 +19,14 @@ if CONFIG['OS_ARCH'] == 'Darwin':
     if '86' in CONFIG['OS_TEST'] and CONFIG['OS_TEST'] != 'x86_64':
         DEFINES['MOZ_NEED_LEADING_UNDERSCORE'] = True
 
-if CONFIG['OS_ARCH'] in ('NetBSD', 'OpenBSD', 'GNU'):
+if CONFIG['OS_ARCH'] in ('OpenBSD', 'GNU'):
     if CONFIG['CPU_ARCH'] == 'x86':
         SOURCES += [
             'xptcinvoke_gcc_x86_unix.cpp',
             'xptcstubs_gcc_x86_unix.cpp'
         ]
 
-if CONFIG['OS_ARCH'] in ('Linux', 'FreeBSD') or \
+if CONFIG['OS_ARCH'] in ('Linux', 'FreeBSD', 'NetBSD', 'DragonFly') or \
    CONFIG['OS_ARCH'].startswith('GNU_'):
     if CONFIG['OS_TEST'] == 'x86_64':
         SOURCES += [
